### ISSUE TITLE
Bridge networks with Docker Playground

### ISSUE THREAD

**Issue by msorens** [DanWahlin/NodeExpressMongoDBDockerApp#5]
Following along with your "Docker for Web Developers" course, I have been using Play with Docker rather than installing Docker Desktop on my local machine (only because I am curious to see how much one can do with that web-based environment).
Everything has been working fine so far, but I may have run into a snag with this exercise: Section 7, Communicating Between Docker Containers/Container Networks in Action.

My repro steps, based off of https://github.com/DanWahlin/NodeExpressMongoDBDockerApp/blob/main/node.dockerfile :
```
docker network create --driver bridge isolated_network
docker run -d --net=isolated_network --name mongodb mongo
git clone https://github.com/DanWahlin/NodeExpressMongoDBDockerApp.git
cd NodeExpressMongoDBDockerApp
docker build -f node.dockerfile -t msorens/nodeapp .
docker run -d --net=isolated_network --name nodeapp -p 3000:3000 -v $(pwd)/logs:/var/www/logs msorens/nodeapp
```
Looks like everything is running nominally:
```
$ docker ps
CONTAINER ID   IMAGE             COMMAND                  CREATED         STATUS         PORTS                    NAMES
3d788a8509d7   msorens/nodeapp   "npm start"              6 seconds ago   Up 4 seconds   0.0.0.0:3000->3000/tcp   nodeapp
ab66858cb7c9   mongo             "docker-entrypoint.s…"   9 minutes ago   Up 9 minutes   27017/tcp                mongodb

$ docker network ls
NETWORK ID     NAME               DRIVER    SCOPE
e74d238684bb   bridge             bridge    local
8f6a96265ecf   host               host      local
dc606b58ec0c   isolated_network   bridge    local
181b0fd1cebf   none               null      local

$ docker network inspect isolated_network | jq '.[0] | .Containers'
{
  "3d788a8509d7659dcb87e9c9a16eb39ef671d6f9dafb66d92153457ea4c8b818": {
    "Name": "nodeapp",
    "EndpointID": "ca8ea8f320dc6d0c95a434f61ebc62e53811f8692d6926b608f9ebdd3c15d15c",
    "MacAddress": "02:42:ac:13:00:03",
    "IPv4Address": "172.19.0.3/16",
    "IPv6Address": ""
  },
  "ab66858cb7c97241445bab5439438dd17ec93880fd35399c8943b8931e7a263c": {
    "Name": "mongodb",
    "EndpointID": "dcac3bc8445f558be9aac95a3b0960352754c849e4b3d31f719479233bd5abb4",
    "MacAddress": "02:42:ac:13:00:02",
    "IPv4Address": "172.19.0.2/16",
    "IPv6Address": ""
  }
}
```
The playground eventually gave me the correct port button:

![image](https://user-images.githubusercontent.com/6817500/115589060-44e25b00-a284-11eb-93a2-5bdcc18ef3b0.png)

And I confirmed that via the data from docker:
```
$ docker inspect nodeapp | jq '.[0] | .HostConfig.PortBindings'
{
  "3000/tcp": [
    {
      "HostIp": "",
      "HostPort": "3000"
    }
  ]
}
```
but upon clicking that port button the site is not reachable in the browser:
![image](https://user-images.githubusercontent.com/6817500/115589466-bae6c200-a284-11eb-9663-caf74c91941f.png)

Is there something about bridge networking that is not quite compatible with the docker playground?

**Comment by DanWahlin** [comment]
Hi @msorens. I haven't used Play with Docker for a long time so it's hard to say what the problem could be there. Everything you show looks right.

I just ran the exact steps you list above (thanks for providing the excellent detail there) and everything worked as expected. It's possible they don't support networking but I'm not sure what they do and don't support since it's not a tool I use. 

To debug I'd run `docker logs [node-containerId]` and see what that shows. I'm guessing it's a DB connection issue but you'd have to try it. You can also try shelling into the container and running the dbSeeder.js script to see if any errors come up connecting. The readme has more details there. If you see connection errors then there's definitely something going on with Play with Docker although I'd assume they support networking since it's very standard.

**Comment by msorens** [comment]
Well, this is curious. I launched a fresh Play with Docker session with my exact steps detailed above... and this time it worked perfectly. I did not change a thing. 

(Which makes me think that I am seeing a pattern, considering both this issue and the other one I recently reported (https://github.com/DanWahlin/Docker-for-Web-Developers/issues/6). In each, my first attempt failed but my second attempt succeeded. 🤔 What is it they say? One is not necessarily paranoid if they really are out to get you.... 😉 )

In any event, thanks for the excellent tutorial, Dan--got a couple more sections to go yet!
